src/TelemetryHelper.ts

```diff
@@ -0,0 +1,106 @@
+
+const ORYX_CLI_IMAGE: string = "mcr.microsoft.com/oryx/cli:debian-buster-20230207.2";
+
+const SUCCESSFUL_RESULT: string = "succeeded";
+const FAILED_RESULT: string = "failed";
+
+const BUILDER_SCENARIO: string = "used-builder";
+const DOCKERFILE_SCENARIO: string = "used-dockerfile";
+const IMAGE_SCENARIO: string = "used-image";
+
+const util = new Utility();
+const githubActionsToolHelper = new GithubActionsToolHelper();
+
+export class TelemetryHelper {
+    readonly disableTelemetry: boolean;
+
+    private scenario: string;
+    private result: string;
+    private errorMessage: string;
+    private taskStartMilliseconds: number;
+
+    constructor(disableTelemetry: boolean) {
+        this.disableTelemetry = disableTelemetry;
+        this.taskStartMilliseconds = Date.now();
+    }
+
+    /**
+     * Marks that the task was successful in telemetry.
+     */
+    public setSuccessfulResult() {
+        this.result = SUCCESSFUL_RESULT;
+    }
+
+    /**
+     * Marks that the task failed in telemetry.
+     */
+    public setFailedResult(errorMessage: string) {
+        this.result = FAILED_RESULT;
+        this.errorMessage = errorMessage;
+    }
+
+    /**
+     * Marks that the task used the builder scenario.
+     */
+    public setBuilderScenario() {
+        this.scenario = BUILDER_SCENARIO;
+    }
+
+    /**
+     * Marks that the task used the Dockerfile scenario.
+     */
+    public setDockerfileScenario() {
+        this.scenario = DOCKERFILE_SCENARIO;
+    }
+
+    /**
+     * Marks that the task used the previously built image scenario.
+     */
+    public setImageScenario() {
+        this.scenario = IMAGE_SCENARIO;
+    }
+
+    /**
+     * If telemetry is enabled, uses the "oryx telemetry" command to log metadata about this task execution.
+     */
+    public async sendLogs() {
+        let taskLengthMilliseconds: number = Date.now() - this.taskStartMilliseconds;
+        if (!this.disableTelemetry) {
+            githubActionsToolHelper.info(`Telemetry enabled; logging metadata about task result, length and scenario targeted.`);
+            try {
+                let resultArg: string = '';
+                if (!util.isNullOrEmpty(this.result)) {
+                    resultArg = `--property result=${this.result}`;
+                }
+
+                let scenarioArg: string = '';
+                if (!util.isNullOrEmpty(this.scenario)) {
+                    scenarioArg = `--property scenario=${this.scenario}`;
+                }
+
+                let errorMessageArg: string = '';
+                if (!util.isNullOrEmpty(this.errorMessage)) {
+                    errorMessageArg = `--property errorMessage=${this.errorMessage}`;
+                }
+
+                let args: string[] = [`run`, `--rm`, `${ORYX_CLI_IMAGE}`, `/bin/bash`, `-c`, `oryx telemetry --event-name ContainerAppsGitHubActionV1 ` + `--processing-time ${taskLengthMilliseconds} ${resultArg} ${scenarioArg} ${errorMessageArg}`];
```

Comment:
A function for fetching this `ContainerAppsGitHubActionV1` argument should be created in the tool helper class

Reply:
Updated.